hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestOzoneAtRestEncryption.java

```diff
@@ -493,4 +493,11 @@ private static void assertReadContent(byte[] inputData, byte[] readData,
         offset + " and length " + readData.length,
         inputDataForComparison, readData);
   }
+
+  @Test
+  public void testGetKeyProvider() throws Exception {
+    KeyProvider kp1 = store.getKeyProvider();
+    KeyProvider kp2 = store.getKeyProvider();
+    Assert.assertEquals(kp1, kp2);
+  }
```

Comment:
Can we add another test to check if the key provider gets closed when the RPC client is closed?

Reply:
I had thought about it but the guava cache removal call back is tricky to test. Spend some time in the morning and figure out a way to use mockito spy to test this indirectly.